File: examples/leetcode-out/zig/4/median-of-two-sorted-arrays.zig
const std = @import("std");

fn findMedianSortedArrays(nums1: []const i32, nums2: []const i32) [2]i32 {
	var merged = std.ArrayList(i32).init(std.heap.page_allocator);
	var i = 0;
	var j = 0;
	while ((((i < nums1.len) or j) < nums2.len)) {
		if ((j >= nums2.len)) {
			try merged.append(@as(i32,@intCast(nums1[i])));
			i = (i + 1);
		} else 		if ((i >= nums1.len)) {
			try merged.append(@as(i32,@intCast(nums2[j])));
			j = (j + 1);
		} else 		if ((nums1[i] <= nums2[j])) {
			try merged.append(@as(i32,@intCast(nums1[i])));
			i = (i + 1);
		} else {
			try merged.append(@as(i32,@intCast(nums2[j])));
			j = (j + 1);
		}
	}
	const total = merged.len;
	if (((total % 2) == 1)) {
		return @as(f64, merged[(total / 2)]);
	}
	const mid1 = merged[((total / 2) - 1)];
	const mid2 = merged[(total / 2)];
	return (@as(f64, ((mid1 + mid2))) / 2);
}

pub fn main() void {
}

File: examples/leetcode-out/hs/4/median-of-two-sorted-arrays.hs
module Main where

import Data.Maybe (fromMaybe)


forLoop :: Int -> Int -> (Int -> Maybe a) -> Maybe a
forLoop start end f = go start
  where
    go i | i < end =
            case f i of
              Just v -> Just v
              Nothing -> go (i + 1)
         | otherwise = Nothing

avg :: Real a => [a] -> Double
avg xs | null xs = 0
      | otherwise = sum (map realToFrac xs) / fromIntegral (length xs)

_indexString :: String -> Int -> String
_indexString s i =
  let idx = if i < 0 then i + length s else i
  in if idx < 0 || idx >= length s
       then error "index out of range"
       else [s !! idx]

_input :: IO String
_input = getLine


findMedianSortedArrays :: [Int] -> [Int] -> Double
findMedianSortedArrays nums1 nums2 = fromMaybe (0.0) $
    (let merged = [] in (let i = 0 in (let j = 0 in (let total = length merged in case if ((total `mod` 2) == 1) then Just ((merged !! (div total 2))) else Nothing of Just v -> Just v; Nothing -> (let mid1 = (merged !! ((div total 2) - 1)) in (let mid2 = (merged !! (div total 2)) in Just ((((mid1 + mid2)) / 2.0))))))))

main :: IO ()
main = do
    return ()
